### PR TITLE
Fix machine epsilon for double

### PR DESCRIPTION
**Pull request by rstm-sf**
## What does the pull request do?
Fixing inaccuracy in machine epsilon found by @MarchingCube https://github.com/AvaloniaUI/Avalonia/pull/4092#discussion_r437634098

## What is the current behavior?
The result of methods from the `MathUtilities` class using machine epsilon was not always correct.

## What is the updated/expected behavior with this PR?
The result of methods from the `MathUtilities` class using machine epsilon.

## Checklist

- [x] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

**Comment by danwalmsley** [comment]
@rstm-sf is this something we should also upstream to dotnet? so that double.Epsilon is correct in the framework?
*disclaimer* I have no idea what this is ;)

**Comment by jmacato** [comment]
@rstm-sf i think this PR could be expanded elsewhere in the whole repo since we use double.Epsilon alot.

File: src/Avalonia.Controls/Slider.cs [hunk 1]
```diff
@@ -193,7 +193,7 @@ private void MoveToPoint(PointerPoint x)
             var orient = Orientation == Orientation.Horizontal;
 
             var pointDen = orient ? _track.Bounds.Width : _track.Bounds.Height;
-            pointDen += double.Epsilon; // Just add epsilon to avoid divide by zero exceptions.
+            pointDen += MathUtilities.DoubleEpsilon; // Just add epsilon to avoid divide by zero exceptions.
```
Comment (rstm-sf) [inline src/Avalonia.Controls/Slider.cs:196]: https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQGUCWBbADgGwgAUEB7AcwSm2QBoYQE4A7AHwGNTnkYACAE1JxghXgA9eAXl4AGAHQyA3AFgAUAAEAjHK0BOABSbeAenEBKFRu17DJ3vokBqAUJEQ5AUVzJM+LmYs1LR1NAyNTB15nACY5aOiZABZEgDYk3WiAVhkAZk08iFQZTRSAxSA

**Comment by rstm-sf** [comment]
> 
> 
> @rstm-sf is this something we should also upstream to dotnet? so that double.Epsilon is correct in the framework?
> _disclaimer_ I have no idea what this is ;)

[This](https://stackoverflow.com/a/32448365) good answer what is double.Epsilon ;)

File: src/Avalonia.Controls/Slider.cs (hunk 1 above)
Comment (rstm-sf) [reply src/Avalonia.Controls/Slider.cs:196]: https://github.com/AvaloniaUI/Avalonia/blob/954c70257e10ceb42a8e80b5ad1dec49256648ce/src/Avalonia.Controls/Slider.cs#L196-L199

https://github.com/AvaloniaUI/Avalonia/blob/954c70257e10ceb42a8e80b5ad1dec49256648ce/src/Avalonia.Base/Utilities/MathUtilities.cs#L101-L115

Comment (MarchingCube) [reply src/Avalonia.Controls/Slider.cs:196]: You are right, I forgot that IEEE spec defines floating point division by zero as producing infinity. As long as our clamp function deals fine with it we won't need that extra epsilon.

Only integer divisions throw exceptions.

**Comment by MarchingCube** [comment]
Problem is that almost all of our calculations are done using double precision (I am still not sure why WPF is using double and not single precision), so exposing single precision API will both force users to cast and potentially introduce truncation errors.

**Comment by rstm-sf** [comment]
I mean, it’s better to add similar checks for single precision ;)

**Review by MarchingCube — APPROVED** [review]
LGTM.

**Pull request merged**